server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/TaskSerializationModule.java

```diff
@@ -56,6 +60,11 @@ protected void configure() {
         return FullReindexingTask.module(performer);
     }
 
+    @ProvidesIntoSet
+    public TaskDTOModule<?, ?> deleteMailsFromMailQueueTask(MailQueueFactory<?> mailQueueFactory) {
+        return DeleteMailsFromMailQueueTaskDTO.module((MailQueueFactory<ManageableMailQueue>) mailQueueFactory);
```

Comment:
what makes you think this assumption stands?

Reply:
all the injection code is based on this assumption :/

Reply:
I don't understand why we can't ask for the actual type in the method signature

Reply:
see https://issues.apache.org/jira/browse/JAMES-2926